Userland/Libraries/LibCrypto/Hash/SHA2.cpp

```diff
@@ -5,9 +5,19 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/Platform.h>
 #include <AK/Types.h>
 #include <LibCrypto/Hash/SHA2.h>
 
+#if (ARCH(I386) || ARCH(X86_64)) && !defined KERNEL && defined(AK_COMPILER_GCC)
+#    include <AK/SIMD.h>
+#    include <AK/SIMDExtras.h>
+#    define SHA256_ATTRIBUTE_TARGET_DEFAULT __attribute__((target("default")))
+#    define SHA256_ATTRIBUTE_TARGET_X86 __attribute__((target("sha"), used))
+#else
+#    define SHA256_ATTRIBUTE_TARGET_DEFAULT
+#endif
+
```

Comment:
These lines shouldn't be duplicated

Reply:
Cannot see any duplication. Can you explain what do you mean? There is conditional definition of `SHA256_ATTRIBUTE_TARGET_DEFAULT` macro, it is needed for non-x86 architectures to have different value.

Reply:
These macros differ from ones in SHA1.cpp only in name.

Reply:
OK, what do you suggest instead? Because this pattern could repeat in the future. Fox example for SHA-512 (when Arrow Lake and Lunar Lake came out) or for AES and VAES. That's what comes to my mind. Maybe also for CRC. And maybe also for different architectures. But each time there will be (possible) different set of conditions.

Reply:
If I were you, I would add something like
```cpp
#define AK_ATTRIBUTE_TARGET_DEFAULT [[gnu::target("default")]]

// FIXME: Clang doesn't support function multiversioning based on "sha" feature.
#if (ARCH(I386) || ARCH(X86_64)) && defined(AK_COMPILER_GCC)
#    define AK_ATTRIBUTE_TARGET_X86_SHA [[gnu::target("sha"), gnu::used]]
#endif
```
to `SIMD.h` / `SIMDExtras.h` and then guard target-specific definitions of `transform` by `#ifdef AK_ATTRIBUTE_TARGET_X86_SHA`

Reply:
I've also looked up why clang doesn't support said multiversioning and the situation seems quite hopeless: https://github.com/llvm/llvm-project/issues/59040, https://discourse.llvm.org/t/proposal-split-built-ins-from-the-rest-of-compiler-rt/67978. So, unfortunately, we might want to consider a different approach some time in the future.

Reply:
it works on clang trunk though
https://godbolt.org/z/oYa515e9G
(I have mentioned this on discord as well I think)